tests/base/src/test/java/org/keycloak/tests/db/CaseSensitiveSchemaTest.java

```diff
@@ -1,70 +1,45 @@
 package org.keycloak.tests.db;
 
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
-import org.keycloak.admin.client.resource.RolesResource;
-import org.keycloak.representations.idm.RoleRepresentation;
-import org.keycloak.testframework.annotations.InjectClient;
 import org.keycloak.testframework.annotations.InjectTestDatabase;
 import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
-import org.keycloak.testframework.config.Config;
+import org.keycloak.testframework.database.DatabaseConfig;
 import org.keycloak.testframework.database.DatabaseConfigBuilder;
 import org.keycloak.testframework.database.PostgresTestDatabase;
 import org.keycloak.testframework.database.TestDatabase;
 import org.keycloak.testframework.injection.LifeCycle;
-import org.keycloak.testframework.realm.ManagedClient;
 import org.keycloak.testframework.server.KeycloakServerConfig;
 import org.keycloak.testframework.server.KeycloakServerConfigBuilder;
-import org.keycloak.testsuite.util.RoleBuilder;
 
 @DisabledIfEnvironmentVariable(named = "KC_TEST_DATABASE", matches = "mssql", disabledReason = "MSSQL does not support setting the default schema per session")
-@KeycloakIntegrationTest(config = CaseSensitiveSchemaTest.KeycloakConfig.class)
-public class CaseSensitiveSchemaTest {
-    @InjectTestDatabase(lifecycle = LifeCycle.CLASS, config = DatabaseConfigurator.class)
-    TestDatabase db;
-
-    @InjectClient
-    ManagedClient managedClient;
+@KeycloakIntegrationTest(config = CaseSensitiveSchemaTest.CaseSensitiveServerConfig.class)
+public class CaseSensitiveSchemaTest extends AbstractDBSchemaTest {
 
-    @Test
-    public void testCaseSensitiveSchema() {
-        RoleRepresentation role1 = RoleBuilder.create()
-              .name("role1")
-              .description("role1-description")
-              .singleAttribute("role1-attr-key", "role1-attr-val")
-              .build();
-
-        RolesResource roles = managedClient.admin().roles();
-        roles.create(role1);
-        roles.deleteRole(role1.getName());
-    }
-
-    protected static String dbType() {
-        String database = Config.getSelectedSupplier(TestDatabase.class);
-        return database == null ? "dev-mem" : database;
-    }
+    @InjectTestDatabase(lifecycle = LifeCycle.CLASS, config = CaseSensitiveDatabaseConfig.class)
```

Comment:
Doesn't strictly need to be LifeCycle.CLASS, as the test framework should figure out that the DB is not compatible and restart things as needed.